mycore-base/src/main/java/org/mycore/services/fieldquery/MCRQuery.java

```diff
@@ -1,221 +1,259 @@
-            query.setAttribute("maxResults", String.valueOf(maxResults));
-
-            if (sortBy != null && sortBy.size() > 0) {
-                Element sortByElem = new Element("sortBy");
-                query.addContent(sortByElem);
-                for (MCRSortBy sb : sortBy) {
-                    Element ref = new Element("field");
-                    ref.setAttribute("name", sb.getFieldName());
-                    ref.setAttribute("order", sb.getSortOrder() ? "ascending" : "descending");
-                    sortByElem.addContent(ref);
-                }
-            }
-
-            Element conditions = new Element("conditions");
-            query.addContent(conditions);
-            conditions.setAttribute("format", "xml");
-            conditions.addContent(cond.toXML());
-            doc = new Document(query);
-        }
-        return doc;
-    }
-
-    /**
-     * Parses a XML representation of a query.
-     * 
-     * @param doc
-     *            the XML document
-     * @return the parsed MCRQuery
-     */
-    public static MCRQuery parseXML(Document doc) {
-        Element xml = doc.getRootElement();
-        Element conditions = xml.getChild("conditions");
-        MCRQuery query = null;
-        if (conditions.getAttributeValue("format", "xml").equals("xml")) {
-            Element condElem = (Element) conditions.getChildren().get(0);
-            query = new MCRQuery(new MCRQueryParser().parse(condElem));
-        } else {
-            String queryString = conditions.getTextTrim();
-            query = new MCRQuery(new MCRQueryParser().parse(queryString));
-        }
-
-        String max = xml.getAttributeValue("maxResults", "");
-        if (max.length() > 0) {
-            query.setMaxResults(Integer.parseInt(max));
-        }
-
-        List<MCRSortBy> sortBy = null;
-        Element sortByElem = xml.getChild("sortBy");
-
-        if (sortByElem != null) {
-            List children = sortByElem.getChildren();
-            sortBy = new ArrayList<MCRSortBy>(children.size());
-
-            for (Object aChildren : children) {
-                Element sortByChild = (Element) aChildren;
-                String name = sortByChild.getAttributeValue("name");
-                String ad = sortByChild.getAttributeValue("order");
-
-                boolean direction = "ascending".equals(ad) ? MCRSortBy.ASCENDING : MCRSortBy.DESCENDING;
-                sortBy.add(new MCRSortBy(name, direction));
-            }
-        }
-        if (sortBy != null) {
-            query.setSortBy(sortBy);
-        }
-
-        return query;
-    }
+	/** The query condition */
+	private MCRCondition cond;
+
+	/** The maximum number of results, default is 0 = unlimited */
+	private int maxResults = 0;
+
+	/** A list of MCRSortBy criteria, may be empty */
+	private List<MCRSortBy> sortBy = new ArrayList<>();
+
+	/** A CSV-list as String of SOLR fields they should be return in response */
+	private String returnFields = "";
```

Comment:
Why not List<String> instead of a single String with comma-separated values?

Reply:
Why should I use a list? As input it is a CSV-list in the search form ( <returnFields>id,objectType</returnFields> ). As output we need a CSV-list like fl=id,objectType . Why should I transform it in a list and then transform it back?